### PR TITLE
packages: add script and shortcode

### LINKED ISSUE #14: document software versions for each release
Create an initial template followed by a maintainable page to document versions of softwares in each release.

### PR DESCRIPTION
**Pull request by mchaker**
**Issue number:**

Closes #14 

**Description of changes:**

Adds a modified version of [Kyle's script](https://github.com/bottlerocket-os/project-website/issues/14#issuecomment-1270245319) that extracts software versions from a Bottlerocket OS repository and outputs them to a Hugo page. Documentation for that script is included.

Also adds the required Hugo shortcode layout to render a table of the software versions.

Software version table pages will live as `index.markdown` files inside a subdirectory under the new `content/en/os/MINOR_RELEASE/packages/` directory. The subdirectory name would be the patch release version e.g. `1.13.0`, `1.13.1`, etc.

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

File: scripts/software-versions-inventory/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+1. Clone the [Bottlerocket SDK repository](https://github.com/bottlerocket-os/bottlerocket-sdk)
+2. `cd` into the directory where you cloned the SDK
+3. Install the Bottlerocket license tool with `cargo install bottlerocket-license-tool --path ./license-tool/`
+
+### Fetching licenses
+
+1. Place a file named `Licenses.toml` in the root of your Bottlerocket OS repository with the following contents:
+
+```toml
```
Comment (stockholmux) [inline scripts/software-versions-inventory/README.md:28]: Is there a way to not have to put in these faux license files? E.g. handle the error produced by the nvidia pacakging?

File: scripts/software-versions-inventory/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Install `cargo` by installing the Rust toolchain via [rustup](https://rustup.rs).
+
+### `make`
+
+If you do not have it installed, install `make` on your system.
+On fedora, run `sudo dnf install make`.
+
+### `rpmspec`
+
+On fedora, run `sudo dnf install rpm-build` to install the `rpmspec` command, which is used in the software versions inventory script.
```
Comment (stockholmux) [inline scripts/software-versions-inventory/README.md:16]: IIRC, I was able to run this on a mac. Maybe instructions for brew? (https://formulae.brew.sh/formula/rpm ? it's been a while)

File: scripts/software-versions-inventory/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+2. Copy `Licenses.toml` to `~/rpmbuild/BUILD/`.
+
+## Running the script
+
+1. Run `./software-versions-inventory.sh PATH_TO_BOTTLEROCKET_OS_REPO > packages.markdown`, where `PATH_TO_BOTTLEROCKET_OS_REPO` is the path to the Bottlerocket OS repository from which you want to extract package versions.
+
+## Script output
+
+The output of the script is a markdown Hugo page containing YAML [front matter](https://gohugo.io/content-management/front-matter/) listing packages and their versions, usable in the Bottlerocket project website.
```
Comment (stockholmux) [inline scripts/software-versions-inventory/README.md:54]: Include a copy of what the generated page source should look like in the readme.

**Review by stockholmux — CHANGES_REQUESTED** [review]
A few notes. I think we can make this more ergonomic!

File: scripts/software-versions-inventory/README.md (hunk 2 above)
Comment (mchaker) [reply scripts/software-versions-inventory/README.md:16]: Added `brew` instructions

File: scripts/software-versions-inventory/README.md (hunk 3 above)
Comment (mchaker) [reply scripts/software-versions-inventory/README.md:25]: Good point, added

File: scripts/software-versions-inventory/README.md (hunk 1 above)
Comment (mchaker) [reply scripts/software-versions-inventory/README.md]: The solution seems to be skipping packages that call out to `bottlerocket-license-tool`. Adding a note about that

**Review by stockholmux — APPROVED** [review]
LGTM - good work on this @mchaker !

**Pull request merged**